packages/server-core-integration/src/lib/gatewaySetup.ts

```diff
@@ -0,0 +1,215 @@
+import * as Winston from 'winston'
+
+/**
+ * Used when JSON.stringifying values that might be circular
+ * Usage: JSON.stringify(value, JSONStringifyCircular()))
+ */
+function JSONStringifyCircular() {
+	const cacheValues: any[] = []
+	const cacheKeys: any[] = []
+	const stringifyFixer = (key: string, value: any) => {
+		if (typeof value === 'object' && value !== null) {
+			const i = cacheValues.indexOf(value)
+			if (i !== -1) {
+				// Duplicate reference found
+				try {
+					// If this value does not reference a parent it can be deduped
+					return JSON.parse(JSON.stringify(value))
+				} catch (error) {
+					// discard key if value cannot be deduped
+					return '[circular of ' + (cacheKeys[i] || '*root*') + ']'
+				}
+			}
+			// Store value in our collection
+			cacheValues.push(value)
+			cacheKeys.push(key)
+		}
+		return value
+	}
+	return stringifyFixer
+}
+const { printf } = Winston.format
+const myLogFormat = Winston.format.combine(
+	// Winston.format.errors({ stack: true }),
+	// Winston.format.metadata(),
+	// Winston.format.json()
+	// Winston.format.simple()
+	printf((obj): string => {
+		if (obj instanceof Error) return obj.stack || obj.toString()
+		return JSON.stringify(obj, JSONStringifyCircular())
+	})
+)
+
+/**
+ * Intended to be run first in the Gateways.
+ * Sets up logging, error handling etc.
+ */
+export function setupGatewayProcess(options: { logPath: string; logLevel: string | undefined }) {
+	let logger: Winston.Logger
+	if (options.logPath) {
```

Comment:
To continue our talk from Tuesday about trying to make smaller functions.
This if-statement presents a good opportunity for that.

As far as I can see the body of the if- and else-statment only uses the variable `logger` and the parameter `options` and the purpose of the bodies is to create the logger. So you could create two methods (one for each of the bodies) that each creates a logger:

```
export function setupGatewayProcess(options: {...}) {
  const logger = options.logPath 
            ? createLoggerWithPath(options.logLevel, options.logPath) 
            : createLogger(options.logLevel)
  // Rest of your function
}

function createLoggerWithPath(...) {
  // The body of your if-statement
}

function createLogger(...) {
  // The body of your else statement
} 
```

Now, as a maintenance developer, I don't have to first read through the code of creating the logger, if my current task has nothing to do with the creation of the logger.

Disclaimer: My naming of the methods might not be accurate. I haven't spent the time to understand the code 100%